app/services/displayable_pii_formatter.rb

```diff
@@ -0,0 +1,86 @@
+class DisplayablePiiFormatter
+  FormattedPii = Struct.new(
+    :email, :all_emails, :verified_at, :x509_subject, :x509_issuer, :full_name,
+    :social_security_number, :address, :birthdate, :phone, keyword_init: true
+  )
+
+  attr_reader :current_user
+  attr_reader :pii
+
+  def initialize(current_user:, pii:)
+    @current_user = current_user
+    @pii = pii
+  end
+
+  def format
+    formatted_pii_hash = {
+      email: email,
+      all_emails: all_emails,
+      verified_at: verified_at,
+      x509_subject: x509_subject,
+      x509_issuer: x509_issuer,
+    }
+    formatted_pii_hash.merge!(formatted_ial2_pii) if pii.any?
+    FormattedPii.new(formatted_pii_hash)
+  end
```

Comment:
fewer hashes!!! I'd consider writing like this:
```suggestion
  # @return [FormattedPii]
  def format
    formatted_pii_hash.merge!(formatted_ial2_pii) if pii.any?
    FormattedPii.new(
      email: email,
      all_emails: all_emails,
      verified_at: verified_at,
      x509_subject: x509_subject,
      x509_issuer: x509_issuer,
    ).tap do |formatted_pii|
      if pii.any?
        formatted_pii.full_name = full_name
        formatted_pii.social_security_number = social_security_number
        formatted_pii.address = address
        formatted_pii.birthdate = dob
        formatted_pii.phone = phone
      end
    end
  end
```
could even remove the `formatted_ial2_pii` method